test/server.js

```diff
@@ -0,0 +1,250 @@
+        }
+
+        // Since our fake environment creates variable ports, we need to ensure that our pagerefs and URLs in the payload
+        // are valid URLs.
+        assert.strictEqual(new URL(entry.pageref).hostname, '127.0.0.1', 'Entry `pageref` should point to localhost.');
+        assert.strictEqual(
+          new URL(entry.request.url).hostname,
+          '127.0.0.1',
+          'Entry `request.url` should point to localhost.'
+        );
+      })
+    );
+
+    const fixture = harFixtures[req.body[0].group.id];
+    let expected;
+    let end;
+    if (req.body[0].group.id === 'buffered-requests') {
+      // If we're testing buffered payloads, let's split our fixture up into chunks based on which buffered positon
+      // we're currently processing. Also emit a signal back to the open socket informing the SDK test that we've
+      // handled this instance of its buffered/staggered payload.
+      end = bufferedSet * 5;
+      expected = fixture.slice(end - 5, end);
+
+      res.io.emit('sdk:metrics:buffer', {
+        position: bufferedSet,
+      });
+
+      if (bufferedSet < 4) {
+        bufferedSet += 1;
+      } else {
+        // Since this server runs for all tests, reset our buffered position back to the start once we've exhausted our
+        // four slices so other tests that run after this that test buffering will work.
+        bufferedSet = 1;
+      }
+
+      // Add a bit of latency into this call so the SDK that's buffering requests can expect some real-world latency
+      // between its buffered calls.
+      await new Promise(resolve => setTimeout(resolve, 1000));
+    } else {
+      expected = fixture;
+    }
+
+    // Update our test fixture to allow us to dynamically do assertions on dates, times, and variable headers coming
+    // from our different SDK test servers. We need to use `expect` instead of `assert` for this assertion because the
+    // expect module allow us to dynamic matching on deep objects with `expect.any()`.
+    expected.forEach((ex, i) => {
+      expected[i]._id = expect.any(String);
+
+      expected[i].request.log.creator.comment = expect.any(String);
+      expected[i].request.log.creator.name = sdkName;
+      expected[i].request.log.creator.version = sdkVersion;
+
+      expected[i].request.log.entries.forEach((ent, ii) => {
+        expected[i].request.log.entries[ii].pageref = expect.any(String);
+        expected[i].request.log.entries[ii].request.headers = expect.any(Array);
+        expected[i].request.log.entries[ii].request.url = expect.any(String);
+        expected[i].request.log.entries[ii].response.headers = expect.any(Array);
+        expected[i].request.log.entries[ii].startedDateTime = expect.any(String);
+        expected[i].request.log.entries[ii].time = expect.any(Number);
+      });
+    });
+
+    try {
+      await expect(req.body).toStrictEqual(expected);
+    } catch (err) {
+      if (err.matcherResult && err.matcherResult.name !== 'toStrictEqual') {
```

Comment:
nit can we simplify to

```javascript
if (err.matcherResult?.name === 'toStrictEqual') {
  throw new AssertionError(...)
}
throw error;
```

Also why do we specifically want an assertion error for that case? A code comment here might be helpful

Reply:
We'd need to cut off support for Node 10 and 12 to use optional chaining syntax unfortunately.